test/xUnit/csharp/test_PowerShell.cs

```diff
@@ -0,0 +1,167 @@
+            }
+        }
+
+        [Fact]
+        public static async Task TestPowerShellInvokeAsyncInUnopenedRunspace()
+        {
+            using (var rs = RunspaceFactory.CreateRunspace())
+            using (var ps = PowerShell.Create(rs))
+            {
+                var results = await ps.AddScript(@"@(1..10).foreach{Start-Sleep -Milliseconds 500}")
+                                      .InvokeAsync();
+
+                Assert.True(results.IsFaulted);
+                Assert.IsType<AggregateException>(results.Exception);
+                Assert.IsType<InvalidRunspaceStateException>(results.Exception.InnerException);
+                Assert.Equal(results.Exception.InnerException.CurrentState, RunspaceState.BeforeOpen);
+                Assert.Equal(results.Exception.InnerException.ExpectedState, RunspaceState.Opened);
+            }
+        }
+
+        [Fact]
+        public static async Task TestPowerShellInvokeAsyncInBusyRunspace()
+        {
+            using (var rs = RunspaceFactory.CreateRunspace())
+            {
+                rs.Open();
+                using (var ps = PowerShell.Create(rs))
+                {
+                    await ps.AddScript(@"@(1..120).foreach{Start-Sleep -Milliseconds 500}")
+                            .InvokeAsync();
+
+                    int time = 0;
+                    while (rs.RunspaceAvailability != RunspaceAvailability.Busy)
+                    {
+                        if ((time += 100) >= 120000)
+                        {
+                            break
+                        }
+                        Thread.Sleep(100);
+                    }
+
+                    Assert.Equal(rs.RunspaceAvailability, RunspaceAvailability.Busy);
+
+                    using (var ps2 = PowerShell.Create(rs))
+                    {
+                        var results = await ps2.AddScript(@"@(6..10).foreach{Start-Sleep -Milliseconds 500}")
+                                               .InvokeAsync();
+
+                        Assert.True(results.IsFaulted);
+                        Assert.IsType<AggregateException>(results.Exception);
+                        Assert.IsType<PSInvalidOperationException>(results.Exception.InnerException);
+                    }
+
+                    ps.Stop(() => { }, null);
+                }
+            }
+        }
+
+        [Fact]
+        public static async Task TestPowerShellInvokeAsyncMultiple()
+        {
+            var tasks = new List<Task>();
+
+            using (var ps1 = PowerShell.Create())
+            using (var ps2 = PowerShell.Create())
+            {
+                tasks.Add(await ps1.AddScript(@"@(1,3,5,7,9,11,13,15,17,19).foreach{Start-Sleep -Milliseconds 500}")
+                                   .InvokeAsync());
+                tasks.Add(await ps2.AddScript(@"@(2,4,6,8,10,12,14,16,18,20).foreach{Start-Sleep -Milliseconds 500}")
+                                   .InvokeAsync());
```

Comment:
Please change the test script to something simple. The current test script takes a long time to finish.

Reply:
Waiting for general feedback on the need for C# tests for this PR. I'm happy to do both, but want confirmation they are necessary before spending more time on the C# tests.